### ISSUE TITLE
zeronet has already remove the `return` of Michelson

### ISSUE THREAD

**Issue by catsigma** [OCamlPro/liquidity#81]
checkout this document: [http://doc.tzalpha.net/whitedoc/michelson.html](http://doc.tzalpha.net/whitedoc/michelson.html)

The core dev team have already removed the `return` type in Michelson before betanet launch.
This may cause huge changes for liquidity.

@mebsout

**Comment by mebsout** [comment]
Yes, we are aware of it. There is a branch working with the current zeronet here https://github.com/mebsout/liquidity/tree/zeronet but it is still work in progress. The compilation part from Liquidity to Tezos should work though so you can already try it on your examples. It is still missing decompilation for some instructions and an up-to-date documentation. This branch will be merged into the next branch of this repo once it is ready.

**Comment by catsigma** [comment]
I've checked the fork in your repo. Found 1 bug and 2 features missing.
Bug: 
```
type storage_t = {
  a: (nat, (nat, nat) big_map) big_map;
  b: nat;
}
```
This type for storage should cause failure, but the compiler won't.

Features:
1. `mutez` missing, and `tez` should be deprecated.
2. Missing `ADDRESS` instruction.

Thanks for the work.

**Comment by catsigma** [comment]
Thanks for the version 0.3

Still lack of
1. `ADDRESS` instruction
2. `address` literal

**Comment by mebsout** [comment]
`ADDRESS` is `Contract.address`

The type `address` is also present, and key hash or contract constants can be type cast to it.

We're keeping `tez` in liquidity as fixed point numbers because having mutez doesn't bring anything to the language and makes things harder to read.

`SENDER` was added as `Current.sender`

We have also added type `byte` with `Bytes.pack` and `Bytes.unpack` but be wary that it cannot be used as keys in maps for instance because of a bug in Michelson/Tezos which will be fixed in a later version of the Tezos node.
